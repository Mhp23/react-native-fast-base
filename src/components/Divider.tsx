import React from 'react';
import {DividerProps} from '../core/types';
import {StyleProp, ViewStyle} from 'react-native';
import {useTheme} from '../hooks';
import Wrapper from './Wrapper';

const defaultProps: DividerProps = {
  width: 0.5,
  mode: 'border',
};

const Divider: React.FC<DividerProps> = ({
  style,
  width,
  mode,
  spaceY,
  spaceTop,
  spaceBottom,
}) => {
  const {colors} = useTheme();

  const dividerStyle: StyleProp<ViewStyle> = [
    mode === 'border' && {
      borderWidth: width,
      borderColor: colors?.flat,
    },
  ];

  return (
    <Wrapper
      style={[dividerStyle, style]}
      {...{
        spaceY,
        spaceBottom,
        spaceTop,
      }}
    />
  );
};

Divider.defaultProps = defaultProps;

export default Divider;
