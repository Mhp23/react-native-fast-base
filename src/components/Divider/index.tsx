import React from 'react';
import {Wrapper} from '..';
import {makeStyle} from '../../utils';
import {StyleSheet} from 'react-native';
import type {ViewStyle} from 'react-native';
import type {DividerProps} from '../../core';
import {useTheme, useStyle} from '../../hooks';
import {useRM} from 'react-native-full-responsive';

const defaultProps: DividerProps = {
  width: 1,
  spaceX: 0,
  spaceY: 0,
  spaceTop: 0,
  mode: 'border',
  spaceBottom: 0,
  direction: 'vertical',
};

const Divider: React.FC<DividerProps> = ({
  mode,
  color,
  style,
  width,
  spaceY,
  spaceX,
  spaceTop,
  direction,
  spaceBottom,
}) => {
  const {colors} = useTheme();

  const {rs} = useRM();

  const dividerStyles = useStyle<ViewStyle>(() => {
    return makeStyle<ViewStyle>({
      marginVertical: rs(spaceY),
      marginHorizontal: rs(spaceX),
      marginTop: rs(spaceTop),
      marginBottom: rs(spaceBottom),
      ...(mode === 'border' && {
        borderColor: color || colors?.flat,
        ...(direction === 'horizontal'
          ? {
              borderStartWidth: rs(width),
            }
          : {
              borderBottomWidth: rs(width),
            }),
      }),
    });
  }, [
    rs,
    mode,
    width,
    color,
    spaceY,
    spaceX,
    spaceTop,
    direction,
    spaceBottom,
    colors?.flat,
  ]);

  return (
    <Wrapper
      testID="FAST_BASE_DIVIDER"
      style={StyleSheet.flatten([dividerStyles, style])}
    />
  );
};

Divider.defaultProps = defaultProps;

export default Divider;
