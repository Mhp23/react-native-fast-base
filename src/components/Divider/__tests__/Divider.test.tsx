import React from 'react';
import Divider from '..';
import {rs} from 'react-native-full-responsive';
import {render, cleanup} from '@testing-library/react-native';

afterEach(cleanup);

describe('Divider component', () => {
  it('should match snapshot', () => {
    const rendered = render(<Divider width={1} />).toJSON();
    expect(rendered).toBeTruthy();
    expect(rendered).toMatchSnapshot();
  });

  it('should render correctly on border mode', () => {
    const rendered = render(<Divider spaceY={10} width={1.5} />);
    const dividerComponent = rendered.getByTestId('FAST_BASE_DIVIDER');
    expect(dividerComponent.props.style).toMatchObject({
      borderBottomWidth: rs(1.5),
      marginVertical: rs(10),
    });
  });

  it('should render correctly on padding mode in vertical direction', () => {
    const rendered = render(<Divider mode="padding" spaceY={10} />);
    const dividerComponent = rendered.getByTestId('FAST_BASE_DIVIDER');
    expect(dividerComponent.props.style).toMatchObject({
      marginVertical: rs(10),
    });
  });

  it('should render correctly on padding mode in horizontal direction', () => {
    const rendered = render(<Divider mode="padding" spaceX={10} />);
    const dividerComponent = rendered.getByTestId('FAST_BASE_DIVIDER');
    expect(dividerComponent.props.style).toMatchObject({
      marginHorizontal: rs(10),
    });
  });
});
