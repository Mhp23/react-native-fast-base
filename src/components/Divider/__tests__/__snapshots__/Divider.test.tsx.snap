// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider component should match snapshot 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 2.0833333333333335,
    }
  }
  testID="FAST_BASE_DIVIDER"
/>
`;
